src/util/Views.h

```diff
@@ -199,6 +200,31 @@ auto inPlaceTransformView(Range&& range, Transformation transformation) {
                                           std::move(transformation));
 }
 
+/// Create a generator the consumes the input generator until it finds a newline
+/// character and the yields views of whole lines.
+inline cppcoro::generator<std::string_view> lineByLine(
+    cppcoro::generator<std::span<std::byte>> generator) {
+  std::string aggregateBuffer{};
+  for (auto span : generator) {
+    static_assert(sizeof(char) == sizeof(std::byte));
+    aggregateBuffer += std::string_view{
+        std::bit_cast<const char*>(&*span.begin()), span.size()};
```

Comment:
This actually is a reinterpret case (which actually is safe, because it is to `const char*`.
Bit casting only the pointer (but not the single characters) is misleading.
What you mean is "Bit-casting and copying every single character, and hoping that the compiler changes it to a single `memcpy` which it does".